classifier/classifier.py

```diff
@@ -1,138 +1,247 @@
-
-    if args.directory is None:
-        directory = getcwd()
-    else:
-        directory = _format_arg(args.directory)
-        if args.output is None:
-            ''' if -d arg given without the -o arg, keeping the files of -d 
-            in the -d path only after classifying '''
-            output = directory
-
-    if args.date:
-        classify_by_date('DD-MM-YYYY', output, directory)
-    else:
-        classify(formats, output, directory)
-
-    sys.exit()
+            self.moveto(file, directory, folder)
+
+        print("Done!")
+        return
+
+    def _format_text_arg(self, arg):
+        if not isinstance(arg, str):
+            arg = arg.decode('utf-8')
+        return arg
+
+    def _format_arg(self, arg):
+        if isinstance(arg, str):
+            arg = self._format_text_arg(arg)
+        return arg
+
+    def main(self):
+        if self.args.version:
+            # Show version information and quit
+            print(VERSION + '\n' + os.path.realpath(__file__))
+            return False
+        if self.args.types:
+            # Show file format information then quit
+            for key, value in self.formats.items():
+                print(key, '\n', value)
+            return False
+        if self.args.edittypes:
+            subprocess.Popen(['xdg-open', CONFIG])
+            return False
+        if bool(self.args.specific_folder) ^ bool(self.args.specific_types):
+            print(
+                'Specific Folder and Specific Types need to be specified together')
+            sys.exit()
+
+        if self.args.specific_folder and self.args.specific_types:
+            specific_folder = self._format_arg(self.args.specific_folder)
+            self.formats = {specific_folder: self.args.specific_types}
+
+        if self.args.output is None:
+            output = os.getcwd()
+        else:
+            output = self._format_arg(self.args.output)
+
+        if self.args.directory is None:
+            directory = os.getcwd()
+        else:
+            directory = self._format_arg(self.args.directory)
+            if self.args.output is None:
+                ''' if -d arg given without the -o arg, keeping the files of -d
+                in the -d path only after classifying '''
+                output = directory
+
+        # Check for a config file in the source file directory
+        if self.args.directory:
+            if os.path.isfile(os.path.join(self.args.directory, DIRCONFFILE)):
+                self.dirconf = os.path.join(self.args.directory, DIRCONFFILE)
+        elif os.path.isfile(os.path.join(os.getcwd(), DIRCONFFILE)):
+            self.dirconf = os.path.join(os.getcwd(), DIRCONFFILE)
+
+        if self.args.date:
+            if self.args.dateformat:
+                self.classify_by_date(self.args.dateformat, output, directory)
+            else:
+                self.classify_by_date(self.dateformat, output, directory)
+        elif os.path.isfile(self.dirconf):
```

Comment:
If self.dirconf is `None`, then it'll throw a TypeError saying NoneType found.
Instead, also check if self.dirconf is not None as:
``` python
elif self.dirconf and os.path.isfile(self.dirconf):
```